src/test/java/CompatibilityTest.java

```diff
@@ -31,6 +33,32 @@ public void v2() throws Exception {
         assertFieldType(i, String[].class, "y");
         assertFieldType(i, Object.class, "z");
     }
+    
+    @Test
+    public void testClassesAreOnlyReWrittenWhenNeeded() throws Exception {
```

Comment:
My suggestion is to add

```java
@Ignore("TODO JENKINS-28799")
```

so that even the WiP PR is safe to merge. Then the fix can include deleting the annotation.